MarsRover/Controller.cs

```diff
@@ -113,23 +78,49 @@ private Direction InitialiseDirection()
         private Location InitialiseLocation()
         {
             _output.WriteLine(Messages.RoverStartLocation);
-            var input = _input.Read();
+            var input = _input.ReadLine();
             var isValidLocation = Validator.IsValidLocation(input, Map.Width, Map.Height);
             while(!isValidLocation)
             {
                 _output.WriteLine(Messages.InvalidInput);
                 _output.WriteLine(Messages.RoverStartLocation);
-                input = _input.Read();
+                input = _input.ReadLine();
                 isValidLocation = Validator.IsValidLocation(input, Map.Width, Map.Height);
             }
             var location = InputParser.ParseLocation(input);
-            // need to check if location contains an obstacle
-            if (Map.HasObstacle(location)) // TODO: is this the correct way to do this??
+
+            if (Map.HasObstacle(location)) 
             {
                 _output.WriteLine(Messages.InvalidInput);

```

Comment:
It may be good to tell the user that there was an obstacle at the entered location i.e. it was valid syntactically but not semantically